### PR TITLE
Implement es2018, fixes #2, #3, #4

### LINKED ISSUE #2: Support ES2018 Unicode property escapes
The grammar itself is pretty straightforward.

To verify the supported list of properties (and their aliases) and values (and their aliases), you could use the following packages:

- https://github.com/mathiasbynens/unicode-match-property-ecmascript
- https://github.com/mathiasbynens/unicode-match-property-value-ecmascript

I keep these packages in sync with the upstream spec, updating them [whenever Unicode releases a new version](https://github.com/tc39/ecma262/issues/1219).

### PR DESCRIPTION
**Pull request by Protryon**
Copied from original implementation in [shift-parser-js #404](https://github.com/shapesecurity/shift-parser-js/pull/404). Fixes #2, #3, #4.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -62,6 +75,53 @@ class PatternAcceptorState {
...
+        this.index = originalIndex;
+        return null;
+      }
+      characterValue = characterValue.value;
+    } else {
+      characterValue = this.pattern.codePointAt(this.index);
+      this.index += String.fromCodePoint(characterValue).length;
+    }
+    let character = String.fromCodePoint(characterValue);
+    // ZWNJ / ZWJå
```
Comment (mathiasbynens) [inline src/index.js]: ZWJå?

Comment (Protryon) [reply src/index.js]: I magically insert these by accident whenever I CMD+S... å

File: src/unicode-properties.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+
```
Comment (mathiasbynens) [inline src/unicode-properties.js:1]: I’d be interested in hearing why you decided against consuming https://github.com/mathiasbynens/unicode-match-property-value-ecmascript.

The current approach seems like a maintenance problem, which could be alleviated by having `src/unicode-properties.js` be auto-generated by a script. But that is pretty much what https://github.com/mathiasbynens/unicode-match-property-value-ecmascript is doing, so `GOTO 0`.

Comment (Protryon) [reply src/unicode-properties.js:1]: Most of the code in this PR was written over 2 months ago in a PR to shift-parser-js, which already had this information available. No need to mix things up, just copy and paste in this case.

Comment (michaelficarra) [reply src/unicode-properties.js:1]: I am okay with adding @mathiasbynens's library as a dependency.

**Comment by michaelficarra** [comment]
@Protryon You'll have to fast-forward the `es2018` branch to the HEAD of `es2016` once your bugfix PRs have been merged.

**Comment by michaelficarra** [comment]
@Protryon Your bugfix PRs have all been merged, can you fast-forward the `es2018` branch to `es2016` so I can see a reasonable diff here?

**Comment by Protryon** [comment]
@michaelficarra I do not have permission to do so, or I would have already. I will open another PR to base this one on though.

**Comment by michaelficarra** [comment]
I don't see any tests that have a non-unicode regexp with a named backreference.

**Comment by Protryon** [comment]
@michaelficarra Added a few.

**Review by michaelficarra — APPROVED** [review]

**Pull request merged**